### ISSUE TITLE
Set without expanding the Expire time

### ISSUE THREAD

**Issue by R4wm** [jellydator/ttlcache#115]
Its not clear to me how to ".Set()" and not start the expire time over again. 
Can someone point to reference or doc or advise ?

**Comment by davseby** [comment]
I think it's not possible. As a workaround, you could get the item and reset it with the modified ttl value.

```go
var ttl time.Duration = cache.DefaultTTL

if previousItem := cache.Get(key); previousItem != nil {
      ttl = previousItem.ExpiresAt().Sub(time.Now())
}

cache.Set(key, value, ttl)
```

**Comment by swithek** [comment]
At present not updating an existing expiration time isn't possible, though it might a good feature for one of the future versions. We could add a new `ttlcache.PreviousTTL` that would indicate that the previous TTL should be set after an update.

**Comment by swithek** [comment]
Implemented in #121
